tests/scripts/all.sh

```diff
@@ -112,6 +112,11 @@ pre_check_environment () {
     fi
 }
 
+if ! [ -f crypto/Makefile ]; then
```

Comment:
I think this belongs to `pre_check_environment()` - ping @gilles-peskine-arm?

Reply:
This should be in `pre_check_git`, since 1. it's related to git and 2. it could be affected by command line options.

Specifically, this check is only necessary when running one of the components that use the crypto submodule, so it would make sense to do it only when running at least one of these components, and make the corresponding part of `cleanup` also depend on whether the crypto submodule is being used. But if you don't want to code the additional complexity of allowing a run without the submodule, that's ok by me.

Reply:
Since we're soon going to turn `USE_CRYPTO_SUBMODULE` on by default, and soonish building without the submodule won't even be possible, I suggest not wasting time here and keeping this test mandatory.

Reply:
I do think you should put it in the right place. That's a cheap change and will reduce headaches later.

Reply:
Yes, sorry for not being clear about that: it should obviously go in the right place. I was only arguing against making it conditional on what components are run.